src/Helpers.php

```diff
@@ -113,13 +113,26 @@ public static function build_url($protocol, $host, $base_path, $url)
             $ret = preg_replace('/\?(.*)$/', "", $ret);
 
             $filepath = realpath($ret);
-            if ($filepath === false) {
-                return null;
+            if ($filepath !== false) {
+                $ret = "$protocol$filepath$res";
+
+                return $ret;
             }
 
-            $ret = "$protocol$filepath$res";
+            if ($url[0] == '/' && !empty($chrootDirs)) {
+                foreach ($chrootDirs as $dir) {
+                    $ret = realpath($dir) . '/' . $url;
```

Comment:
Since we only reach this section when $url starts with '/' this should probably just be:

```suggestion
                    $ret = realpath($dir) . $url;
```